pkg/scheduler/framework/types.go

```diff
@@ -246,7 +246,11 @@ func (f *FitError) Error() string {
 		sort.Strings(reasonStrings)
 		return reasonStrings
 	}
-	reasonMsg := fmt.Sprintf(NoNodeAvailableMsg+": %v.", f.NumAllNodes, strings.Join(sortReasonsHistogram(), ", "))
+	failedPlugins := make(sets.String)
+	for _, status := range f.Diagnosis.NodeToStatusMap {
+		failedPlugins.Insert(status.failedPlugin)
+	}
+	reasonMsg := fmt.Sprintf(NoNodeAvailableMsg+": %v, failed plugins: %v.", f.NumAllNodes, strings.Join(sortReasonsHistogram(), ", "), failedPlugins.List())
```

Comment:
How useful is it to have the list of plugins that failed for each of the nodes?

Maybe it could go in its own log line, per node, with a higher verbosity level. Similar to what we have for scores

https://github.com/kubernetes/kubernetes/blob/9dba11d971dd940b7fe9901f5d723ead8dbf33b8/pkg/scheduler/generic_scheduler.go#L489-L493

Reply:
We had a conversation on generalizing the error msg. A possible option is to concatenate the `failedPlugin` with `reason` - say modify L237 to:

```go
reasons[fmt.Sprintf("[%v] %v", status.failedPlugin, reason)]++
// TODO: status.failedPlugin may be empty
```

cc @damemi

Reply:
> How useful is it to have the list of plugins that failed for each of the nodes?

When I first saw this in the log, 
> Status after running PostFilter plugins for pod" pod="default/mypod1" status=&{code:2 reasons:[0/2 nodes are available: 2 Preemption is not helpful for scheduling.] err:<nil> failedPlugin:}

I was a little misleading as the both the `err` and `failedPlugin` is empty, have the info in the log might help us to know where is the failure point.

Reply:
> with a higher verbosity level.

level is increased to 8, or else nothing is changed.